extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -556,16 +551,41 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
 
     const pageLevelParametersPromise = getPageLevelParameters_(
         this.win, this.getAmpDoc(), startTime);
-    const rtcRequestPromise = isExperimentOn(this.win, 'disable-rtc') ?
-    Promise.resolve({}) : this.executeRtc_();
     return Promise.all(
-      [pageLevelParametersPromise, rtcRequestPromise]).then(values => {
-        return googleAdUrl(
-            this, DOUBLECLICK_BASE_URL, startTime, Object.assign(
-                this.getBlockParameters_(),
-                /* RTC Parameters */ values[1],
-                /* pageLevelParameters */ values[0]));
-      });
+        [pageLevelParametersPromise, opt_rtcResponsesPromise]).then(values => {
+          const pageLevelParameters = values[0];
+          const rtcParams = this.mergeRtcResponses(values[1]);
+          return googleAdUrl(
+              this, DOUBLECLICK_BASE_URL, startTime, Object.assign(
+                  this.getBlockParameters_(), rtcParams, pageLevelParameters));
+        });
+  }
+
+  mergeRtcResponses(rtcResponses) {
+    if (!rtcResponses) {
+      return null;
+    }
+    let rtcParams = {};
+    const artc = [];
+    const ati = [];
+    const ard = [];
+    rtcResponses.forEach(rtcResponse => {
+      artc.push(rtcResponse['rtcTime']);
+      ati.push(!rtcResponse['error'] ? RTC_ATI_ENUM.RTC_SUCCESS :
+               RTC_ATI_ENUM.RTC_FAILURE);
+      ard.push(rtcResponse['callout']);
+      rtcParams = Object.assign(rtcParams, rtcResponse['rtcResponse']);
+    });
+    ['targeting', 'categoryExclusions'].forEach(key => {
+      if (!!rtcParams[key]) {
+        this.jsonTargeting_[key] =
+            !!this.jsonTargeting_[key] ?
+            deepMerge(this.jsonTargeting_[key], rtcParams[key]) :
```

Comment:
I think this code would be more readable if deepMerge could handle getting null parameters.

Reply:
while I don't disagree, I'm not sure if I want to modify deepMerge